libpod/container_internal.go

```diff
@@ -930,17 +930,30 @@ func (c *Container) makeBindMounts() error {
 	// SHM is always added when we mount the container
 	c.state.BindMounts["/dev/shm"] = c.config.ShmDir
 
-	// Make /etc/resolv.conf
-	if _, ok := c.state.BindMounts["/etc/resolv.conf"]; ok {
-		// If it already exists, delete so we can recreate
-		delete(c.state.BindMounts, "/etc/resolv.conf")
-	}
-	newResolv, err := c.generateResolvConf()
-	if err != nil {
-		return errors.Wrapf(err, "error creating resolv.conf for container %s", c.ID())
-	}
-	c.state.BindMounts["/etc/resolv.conf"] = newResolv
+	if !c.config.NetworkDisabled {
+		// Make /etc/resolv.conf
+		if _, ok := c.state.BindMounts["/etc/resolv.conf"]; ok {
+			// If it already exists, delete so we can recreate
+			delete(c.state.BindMounts, "/etc/resolv.conf")
+		}
+		newResolv, err := c.generateResolvConf()
+		if err != nil {
+			return errors.Wrapf(err, "error creating resolv.conf for container %s", c.ID())
+		}
+		c.state.BindMounts["/etc/resolv.conf"] = newResolv
+
+		// Make /etc/hosts
+		if _, ok := c.state.BindMounts["/etc/hosts"]; ok {
+			// If it already exists, delete so we can recreate
+			delete(c.state.BindMounts, "/etc/hosts")
+		}
```

Comment:
Do we need to worry about retaining any entries from /etc/hosts or /etc/resolv.conf if they already exist?  I'm on the fence so I'll throw this next one up for consideration.  Should we have a logrus.debug  inside both if's just before the delete?  Just something to note that the file is about to be deleted.

Reply:
yes I think we should check for that, still an user could specify a custom `/etc/hosts`

Reply:
I think that Docker solves this by making `--net` conflict with all options related to hosts and DNS - so if you join the net namespace of another container, or no net namespace at all, you cannot set custom hosts or DNS configuration